tests/conftest.py

```diff
@@ -81,34 +84,62 @@ def stats(self):
 
     def iter(self, **kwargs):
         start = kwargs.get("start", 0)
+        count = kwargs.get("count", None)
+        counter = 0
         if start:
             start = int(start.split("/")[-1])
-        count = kwargs.get("count", len(self.items) - start)
 
-        # Scrapinghub API returns all posible items even if `count` greater than possible
-        if start + count > len(self.items):
-            limit = len(self.items)
+        if kwargs.get("filter"):
+            field_name = kwargs.get("filter")[0][0]
+            value = kwargs.get("filter")[0][1][0]
+
+            for item in self.items[start:]:
+                if counter == count:
+                    return
+                if item.get(field_name) == value:
+                    counter += 1
+                    yield item
+
         else:
-            limit = start + count
+            for item in self.items[start:]:
+                if counter == count:
+                    return
+                counter += 1
+                yield item
+
+
+class StoreSource:
+    def __init__(self, items: List[Dict] = None):
+        self.items = items
+
+    def count(self):
+        return len(self.items)
+
+    def iter(self, **kwargs):
+        start = kwargs.get("start", None)
+        if start:
+            start_idx = [
+                i for i, item in enumerate(self.items) if item.get("_key") == start
```

Comment:
I'd rewrite it as generator to avoid evaluating all items (not sure if it would be a good performance improvement)
`start_idx = next(i for i, item in enumerate(self.items) if ...)

Reply:
I don't want to overthink it :)